xds/src/main/java/io/grpc/xds/XdsListenerResource.java

```diff
@@ -165,6 +165,12 @@ static EnvoyServerProtoData.Listener parseServerSideListener(
     if (proto.getAddress().hasSocketAddress()) {
       SocketAddress socketAddress = proto.getAddress().getSocketAddress();
       address = socketAddress.getAddress();
+      if (address.trim().isEmpty()) {
+        throw new ResourceInvalidException("Invalid address: Empty address is not allowed.");
+      }
+      if (socketAddress.hasNamedPort()) {
+        throw new ResourceInvalidException("NAMED_PORT is not supported in gRPC.");
```

Comment:
Move this to the `switch`? Or delete the NAMED_PORT case in the switch; it is dead code and misleading.